### ISSUE TITLE
Discussion about training approaches

### ISSUE THREAD

**Issue by Brugio96** [DigitalPhonetics/IMS-Toucan#87]
Hi, I've been doing some runs, and I would like to understand better how to obtain the best results for my objective.

So, I have two small custom english datasets (in total there are two speakers with 40 minutes of data for each one). 
What I would want is to fine tune on those two datasets, such that the newly trained model is able to reproduce those 2 voices with great similarity in the way they speak in english, and then be able to speak new languages with their voices.
So, what is important to me is that the voices in english are good clones, but I want also to retain the capability of the model to speak different languages with the voices that I add while training/fine tuning.

FIRST RUN 
Initially what I did was to simply fine-tune the model on only the 2 english datasets.
Results: the model learns to clone those two voices in english, but the way in which they speak other languages is sensibly worse, as if the model kinda slightly forgets how to speak.

SECOND RUN
Then I tried adding 2 italian single speaker datasets of medium size (3 and 5 hours) and fine tune as before to try to see if the model can retain at least a good italian (as well as english) as long as the data I train on is of both languages.
Results: maybe because of the unbalance of the two languages in the new training set, but I can't get similar enough clones in english, but at least the model knows how to both speak italian and english.

THIRD RUN
For the third run I reduced the italian dataset size to be about the same of those in english and I increased the learning rate.
Results: The model is able to do a decent clone of the english voice and to speak other languages with the timbre and speaker identity of the target speaker. 
What I observed is that some languages are retained more than other: for example spanish seems to be good, but french is quite bad (it almost forgets what french is supposed to be spoken)
Then, another observation: I have in total 4 datasets: 2 italians (one male and one female) and two english (one male and one female). When I try to speak in the voice of the english female I get good voice similarity in all languages except in italian, where the voice in this case is like a mix of the english female and the italian female.
That make me think that even if add all languages to the training dataset, which I was supposed to do on my next run, the voice similarity will get worse as for example it will some kind of average between all the similar female speakers in the dataset, but this is just my guess.

So, finally I would like to ask you @Flux9665 some things:

- What is the best way in your opinion to approach the problem? 
My goal is to add new voices such that they are good clones in their original language (similar timbre but also similar prosody and good quality) and then they are able to speak all the other languages with a good voice similarity as well as good quality of the spoken language itself (no forgetting problem).
Should I add all the languages to the datasets, should I add all the original datasets or also new ones are good? 
Should I just fine tune the PortaSpeechMeta or also the aligner or vocoder?

- Could you please explain what is phase1 and phase2 referred to in the training? What is cycle loss? How it affects the model?

I apologize for the length of my message, but I want to make the most of your excellent work and I believe your response could be helpful to others who are trying to use the same toolkit.

Have a great day 🤟🏻😊

**Comment by Ca-ressemble-a-du-fake** [comment]
Hi I cannot answer your questions but I can give you a link to an answer Florian already made about [cycle objectives](https://github.com/DigitalPhonetics/IMS-Toucan/issues/58). Maybe this can help you better grasp the concept of cycle objectives.
